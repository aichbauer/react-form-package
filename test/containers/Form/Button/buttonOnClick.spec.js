import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  Field,
  Button,
} from '../../../../src';

configure({ adapter: new Adapter() });

test('Render Form (validate) with button | click button', () => {
  const onClickMock = jest.fn();

  const myComponent = (
    <Form
      validate
    >
      <div>
        <div>checkbox</div>
        <Field type="checkbox" id="checkbox" />
      </div>
      <div>
        <Button id="submit" type="submit" onClick={onClickMock}>submit</Button>
      </div>
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);

  wrapper.find('button').simulate('click');
  wrapper.update();

  expect(onClickMock).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});
