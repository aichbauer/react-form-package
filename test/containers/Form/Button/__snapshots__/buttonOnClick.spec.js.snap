// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form (validate) with button | click button 1`] = `
<form>
  <div>
    <div>
      checkbox
    </div>
    <input
      bindToAlways={false}
      checked={false}
      className="rfp-checkbox"
      dynamic={false}
      id="checkbox"
      invalid={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
      valid={false}
      value={false}
    />
  </div>
  <div>
    <button
      className="rfp-button"
      disabled={false}
      id="submit"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      type="submit"
    >
      submit
    </button>
  </div>
</form>
`;
